core/commands/cmdenv/env.go

```diff
@@ -22,13 +23,22 @@ func GetNode(env interface{}) (*core.IpfsNode, error) {
 }
 
 // GetApi extracts CoreAPI instance from the environment.
-func GetApi(env cmds.Environment) (coreiface.CoreAPI, error) {
+func GetApi(env cmds.Environment, req *cmds.Request) (coreiface.CoreAPI, error) {
 	ctx, ok := env.(*commands.Context)
 	if !ok {
 		return nil, fmt.Errorf("expected env to be of type %T, got %T", ctx, env)
 	}
 
-	return ctx.GetAPI()
+	local, _ := req.Options["local"].(bool)
```

Comment:
So, this'll apply this option (correctly) to *all* commands fixing part of https://github.com/ipfs/go-ipfs/issues/4182. I wonder if we should fix the "confusing" part while we're at it by renaming this option to "offline" (keeping a hidden alias for compatibility).

That would also allow us to share the same option with `ipfs daemon --offline` which should help to further reduce confusion.

Reply:
I'm fine with doing that (the hidden alias could throw warnings at people). This will probably need support on the js side too

Reply:
I didn't realize js replicated that option.

@alanshaw how would you feel about switching from `--local` to `--offline` (keeping `--local` as a backcompat alias)?

Reply:
Implemented, waiting for @alanshaw's opinion (assuming js-ipfs implements `--local`)